src/libclipper/src/rpc_service.cpp

```diff
@@ -43,21 +44,28 @@ RPCService::RPCService()
 RPCService::~RPCService() { stop(); }
 
 void RPCService::start(const string ip, const int port) {
+  if(active_.load(std::memory_order_seq_cst)) {
+    throw std::runtime_error("Attempted to start RPC Service when it is already running!");
+  }
   const string address = "tcp://" + ip + ":" + std::to_string(port);
-  active_ = true;
+  active_.store(true, std::memory_order_seq_cst);
```

Comment:
Explicitly calling `store()` is fine, but just so you know, the assignment operators are overloaded for `atomic<bool>` to call the appropriate atomic methods (http://en.cppreference.com/w/cpp/atomic/atomic/operator%3D).